huaweicloud/services/acceptance/fgs/resource_huaweicloud_fgs_trigger_test.go

```diff
@@ -0,0 +1,685 @@
+	)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.TestAccPreCheck(t) },
+		Providers:    acceptance.TestAccProviders,
+		CheckDestroy: rc.CheckResourceDestroy(),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccFunctionGraphSmnTrigger_basic(randName),
+				Check: resource.ComposeTestCheckFunc(
+					rc.CheckResourceExists(),
+					resource.TestCheckResourceAttr(resourceName, "type", "SMN"),
+					resource.TestCheckResourceAttrSet(resourceName, "smn.0.topic_urn"),
+					acceptance.TestCheckResourceAttrWithVariable(resourceName, "function_urn",
+						"${huaweicloud_fgs_function.test.urn}"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccFunctionGraphTrigger_kafka(t *testing.T) {
+	var (
+		randName  = acceptance.RandomAccResourceName()
+		adminPass = fmt.Sprintf("%s%s%d", acctest.RandString(5), acctest.RandStringFromCharSet(2, "#$"),
+			acctest.RandIntRange(100, 999))
+		resourceName = "huaweicloud_fgs_trigger.test"
+		timeTrigger  trigger.Trigger
+	)
+
+	rc := acceptance.InitResourceCheck(
+		resourceName,
+		&timeTrigger,
+		getTriggerResourceFunc,
+	)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.TestAccPreCheck(t) },
+		Providers:    acceptance.TestAccProviders,
+		CheckDestroy: rc.CheckResourceDestroy(),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccFunctionGraphKafkaTrigger_basic(randName, adminPass),
+				Check: resource.ComposeTestCheckFunc(
+					rc.CheckResourceExists(),
+					resource.TestCheckResourceAttr(resourceName, "type", "KAFKA"),
+					resource.TestCheckResourceAttr(resourceName, "kafka.0.batch_size", "100"),
+					resource.TestCheckResourceAttr(resourceName, "kafka.0.topic_ids.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "status", "ACTIVE"),
+					acceptance.TestCheckResourceAttrWithVariable(resourceName, "function_urn",
+						"${huaweicloud_fgs_function.test.urn}"),
+					acceptance.TestCheckResourceAttrWithVariable(resourceName, "kafka.0.instance_id",
+						"${huaweicloud_dms_kafka_instance.test.id}"),
+				),
+			},
+			{
+				Config: testAccFunctionGraphKafkaTrigger_update(randName, adminPass),
+				Check: resource.ComposeTestCheckFunc(
+					rc.CheckResourceExists(),
+					resource.TestCheckResourceAttr(resourceName, "type", "KAFKA"),
+					resource.TestCheckResourceAttr(resourceName, "kafka.0.batch_size", "100"),
+					resource.TestCheckResourceAttr(resourceName, "kafka.0.topic_ids.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "status", "DISABLED"),
+					acceptance.TestCheckResourceAttrWithVariable(resourceName, "function_urn",
+						"${huaweicloud_fgs_function.test.urn}"),
+					acceptance.TestCheckResourceAttrWithVariable(resourceName, "kafka.0.instance_id",
+						"${huaweicloud_dms_kafka_instance.test.id}"),
+				),
+			},
+		},
+	})
+}
+
+func testAccFunctionGraphTimingTrigger_base(rName string) string {
+	return fmt.Sprintf(`
+resource "huaweicloud_fgs_function" "test" {
```

Comment:
seems that we can extract huaweicloud_fgs_function as a base resource.

Reply:
finished